sympy/core/numbers.py

```diff
@@ -1165,6 +1183,153 @@ def __format__(self, format_spec):
 RealNumber = Float
 
 
+class ComplexFloat(Number):
+    """Represent a complex floating-point number of arbitrary precision.
+    """
+    __slots__ = ['real', 'imag']
+
+    # each Float represents many real numbers,
+    # both erational and irrational.
+    is_rational = None
+    is_irrational = None
+    is_number = True
+
+    is_complex = True
+
+    def __new__(cls, real, imag=None, prec=None):
+        # TODO: do something with prec
```

Comment:
Should also accept a string, Python builtin float/complex types, and tuple (like `_mpc_`).